c7n/resources/iam.py

```diff
@@ -380,3 +394,62 @@ def process(self, resources):
                     client.delete_access_key(
                         UserName=r['UserName'],
                         AccessKeyId=k['AccessKeyId'])
+
+
+#################
+#   IAM Groups  #
+#################
+
+
+@Group.filter_registry.register('used')
+class IamGroupWithUsers(Filter):
+
+    schema = type_schema('used')
+
+    def _user_count(self, resource):
+        client = local_session(self.manager.session_factory).client('iam')
+        return len(client.get_group(GroupName=resource['GroupName'])['Users'])
+
+    def process(self, resources, events=None):
+        return [r for r in resources if self._user_count(r) > 0]
+
+
+@Group.filter_registry.register('unused')
+class IamGroupWithoutUsers(Filter):
+
+    schema = type_schema('unused')
+
+    def _user_count(self, resource):
+        client = local_session(self.manager.session_factory).client('iam')
+        return len(client.get_group(GroupName=resource['GroupName'])['Users'])
+
+    def process(self, resources, events=None):
+        return [r for r in resources if self._user_count(r) == 0]
+
+
+@Group.filter_registry.register('inline-policy-used')
+class IamGroupInlinePolicyUnused(Filter):
+
+    schema = type_schema('inline-policy-used')
+
+    def _inline_policies(self, resource):
+        client = local_session(self.manager.session_factory).client('iam')
+        return len(client.list_group_policies(
+            GroupName=resource['GroupName'])['PolicyNames'])
+
+    def process(self, resources, events=None):
+        return [r for r in resources if self._inline_policies(r) > 0]
+
+
+@Group.filter_registry.register('inline-policy-unused')
+class IamGroupInlinePolicyUsed(Filter):
+
```

Comment:
could you doc string on the filters with inline-policy-used/unused.. i'm not really clear what that means. its not really a used in this context, its more of a `has`. For some of the other filters in other resources, we've started doing `has-inline-policy` and instead of negating, just having a value: True default that can be switched to value: False to specify looking for the negation.